bun/components/TodoList.svelte

```diff
@@ -0,0 +1,123 @@
+<script>
+  // Define a Todo type using JSDoc for documentation purposes
+  /**
+   * @typedef {Object} Todo
+   * @property {string} text
+   * @property {boolean} completed
+   */
+
+  // Initialize an empty array for todos
+  /** @type {Todo[]} */
+  let todos = [];
+  let newTodo = "";
+
+  // Function to add a new todo
+  function addTodo() {
+    if (newTodo) {
+      todos = [...todos, { text: newTodo, completed: false }];
+      newTodo = "";
+    }
+  }
+
+  // Function to check and uncheck the to do item
+  function check(index) {
+    todos = todos.map((todo, i) => 
+      i === index ? { ...todo, completed: !todo.completed } : todo
+    );
+  }
+
+  // Function to remove an item from the list of to do items
+  function remove(index) {
+    todos = todos.filter((_, i) => i !== index);
+  }
```

Comment:
```suggestion
  // Function to check and uncheck the to do item
  const check = (index: number) => 
    todos = todos.map((todo, i) => i === index ? { ...todo, completed: !todo.completed } : todo);
  

  // Function to remove an item from the list of to do items
  const remove = (index: number) => todos = todos.filter((_, i) => i !== index);
```

Just another instance of moving to lambdas